### PR TITLE
NPE fix (No identity key)

### PR DESCRIPTION
**Pull request by williamdes**
Fixes an NPE when trying to display the identity key (when user has no identity key) in a QrCode

### CODE REVIEW

**Comment by williamdes** [comment]
@BLeQuerrec can you review this PR when you have time ?

**Comment by BLeQuerrec** [comment]
Yes, but next week. For now, I just saw that there are missing spaces in `} else {` lines.
Additionally, add `Fixes #XXX` if your PR fixes some issues.

File: src/org/smssecure/smssecure/KeyScanningActivity.java [hunk 1]
```diff
@@ -117,8 +117,14 @@ protected void initiateScan() {
   }
 
   protected void initiateDisplay() {
-    IntentIntegrator intentIntegrator = getIntentIntegrator();
-    intentIntegrator.shareText(Base64.encodeBytes(getIdentityKeyToDisplay().serialize()));
+    IdentityKey identityKey = getIdentityKeyToDisplay();
+    if(identityKey != null) {
+      IntentIntegrator intentIntegrator = getIntentIntegrator();
+      intentIntegrator.shareText(Base64.encodeBytes(identityKey.serialize()));
+    }else{
```
Comment (BLeQuerrec) [inline src/org/smssecure/smssecure/KeyScanningActivity.java]: `} else {`

File: src/org/smssecure/smssecure/crypto/KeyExchangeInitiator.java [hunk 2]
```diff
@@ -95,11 +96,16 @@ public static void initiateKeyExchange(Context context, MasterSecret masterSecre
     SessionBuilder    sessionBuilder    = new SessionBuilder(sessionStore, preKeyStore, signedPreKeyStore,
                                                              identityKeyStore, new SignalProtocolAddress(recipient.getNumber(), 1));
 
-    KeyExchangeMessage         keyExchangeMessage = sessionBuilder.process();
-    String                     serializedMessage  = Base64.encodeBytesWithoutPadding(keyExchangeMessage.serialize());
-    OutgoingKeyExchangeMessage textMessage        = new OutgoingKeyExchangeMessage(recipients, serializedMessage, subscriptionId);
+    if(identityKeyStore.getIdentityKeyPair() != null) {
```
Comment (BLeQuerrec) [inline src/org/smssecure/smssecure/crypto/KeyExchangeInitiator.java]: `if (`

File: src/org/smssecure/smssecure/crypto/KeyExchangeInitiator.java [hunk 3]
```diff
@@ -95,11 +96,16 @@ public static void initiateKeyExchange(Context context, MasterSecret masterSecre
...
+      KeyExchangeMessage         keyExchangeMessage = sessionBuilder.process();
+      String                     serializedMessage  = Base64.encodeBytesWithoutPadding(keyExchangeMessage.serialize());
+      OutgoingKeyExchangeMessage textMessage        = new OutgoingKeyExchangeMessage(recipients, serializedMessage, subscriptionId);
 
-    MessageSender.send(context, masterSecret, textMessage, -1, false);
+      MessageSender.send(context, masterSecret, textMessage, -1, false);
+    }else{
```
Comment (BLeQuerrec) [inline src/org/smssecure/smssecure/crypto/KeyExchangeInitiator.java]: `} else {`

**Comment by williamdes** [comment]
I fixed my indentation problems.
Do I need to open an issue for this PR. I created this PR to fix the 2 NPEs I had when my version of the app had no identity.
Will you use [Prettier](https://github.com/prettier/prettier) with the java plugin when it will be stable ?

**Comment by BLeQuerrec** [comment]
No need to open an issue, but I'm pretty sure someone reported something like that.
There's no plan to use Prettier.

**Comment by williamdes** [comment]
Ok, I will quickly look into the content of the open and closed issues.

**Comment by williamdes** [comment]
@BLeQuerrec can you review this PR ?

**Pull request merged**